astropy/visualization/wcsaxes/wcsapi.py

```diff
@@ -0,0 +1,254 @@
+# Functions/classes for WCSAxes related to APE14 WCSes
+
+import numpy as np
+
+from astropy.coordinates import SkyCoord, ICRS
+from astropy import units as u
+from astropy.wcs import WCS
+from astropy.wcs.wcsapi import SlicedLowLevelWCS
+
+from .frame import RectangularFrame, EllipticalFrame
+from .transforms import CurvedTransform
+
+__all__ = ['transform_coord_meta_from_wcs', 'WCSWorld2PixelTransform',
+           'WCSPixel2WorldTransform']
+
+IDENTITY = WCS(naxis=2)
+IDENTITY.wcs.ctype = ["X", "Y"]
+IDENTITY.wcs.crval = [0., 0.]
+IDENTITY.wcs.crpix = [1., 1.]
+IDENTITY.wcs.cdelt = [1., 1.]
+
+# TODO: Make a ReversedPixelWCS thingy
+
+
+def transform_coord_meta_from_wcs(wcs, frame_class, aslice=None):
+
+    is_fits_wcs = isinstance(wcs, WCS)
+
+    coord_meta = {}
+    coord_meta['name'] = []
+    coord_meta['type'] = []
+    coord_meta['wrap'] = []
+    coord_meta['unit'] = []
+    coord_meta['format_unit'] = []
+
+    invert_xy = False
+    if aslice is not None:
+        wcs_slice = list(aslice)
+        wcs_slice[wcs_slice.index("x")] = slice(None)
+        wcs_slice[wcs_slice.index("y")] = slice(None)
+        wcs = SlicedLowLevelWCS(wcs, wcs_slice[::-1])
+        invert_xy = aslice.index('x') > aslice.index('y')
+
+    transform = WCSPixel2WorldTransform(wcs, invert_xy=invert_xy)
+
+    for idx in range(wcs.world_n_dim):
+
+        axis_type = wcs.world_axis_physical_types[idx]
+        axis_unit = u.Unit(wcs.world_axis_units[idx])
+        coord_wrap = None
+        format_unit = axis_unit
+
+        coord_type = 'scalar'
+
+        if axis_type is not None:
+
+            axis_type_split = axis_type.split('.')
+
+            if "pos.helioprojective.lon" in axis_type:
+                coord_wrap = 180.
+                format_unit = u.arcsec
+                coord_type = "longitude"
+            elif "pos.helioprojective.lat" in axis_type:
+                format_unit = u.arcsec
+                coord_type = "latitude"
```

Comment:
@yashrsharma44 We should also add heliographic coordinates to this list (and for that matter to NDCube).